app/models/goal.py

```diff
@@ -3,4 +3,20 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship('Task', backref='goal', lazy=True)
+
+    # This instance method returns a JSON dictionary of attribute values 
+    # with an optional argument if the goal has tasks assigned to it
+    def convert_to_json(self, tasks_list=None):
+
+        response_body = {  
+            "id": self.goal_id,
+            "title": self.title
+        }
+
+        if tasks_list != None:
+            response_body["tasks"] = tasks_list
```

Comment:
By changing the code to make use of the tasks attribute you added to the Goal (line 8), you guarantee the list of tasks is always exactly what the database says it is, rather than allowing the caller of the function to specify whatever tasks they want to pass in.
Happy to talk more about this if this isn't clear!
(I should note, this will be slightly different behavior that what the current code does because it'll display an empty list for "tasks" when no tasks are present instead of omitting "tasks" from the response.)
```suggestion
        response_body = {  
            "id": self.goal_id,
            "title": self.title,
            "tasks": self.tasks
        }
```

Reply:
That's what I thought - but if I make this change, it causes my Wave 5/6 tests to fail? And I remember  trying something like that initially and not being able to get it to work, which is why I went this route. I'm guessing something else in my code needs to be changed as well, but I'm having trouble identifying where. But I had to use  tasks = Task.query.filter_by(match_goal_id=goal_id) as opposed to just treating tasks like an attribute of a goal, which was not working

Reply:
You're totally right! This suggestion is not enough on its own. Sorry for the confusion and thanks for the thorough response! 
There would need to be more code to convert the data retrieved into the appropriate format.

As far as the other confusion about needing to query manually, I'm not sure. I just made a change in your code (I'll add another comment to talk about it) and it passed the tests just fine.

For all of this though, I want to note that these are quite minor comments. The whole discussion we're having here can be fun and interesting to look into but none of it is very important or of high concern!